src/main/java/seedu/address/storage/JsonAdaptedModule.java

```diff
@@ -113,6 +113,11 @@ public Module toModelType() throws IllegalValueException {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     GradeTracker.class.getSimpleName()));
         }
+        try {
+            gradeTracker.toModelType();
+        } catch (IllegalValueException illegalValueException) {
```

Comment:
I think catching and throwing the same exception here is redundant, as long as the method caller is handling the correct exception, we can omit this try catch block.

Reply:
Thanks, noted!